### DISCUSSION TITLE
easy one click run as administrator in macOS

### DISCUSSION THREAD

**Discussion by mostlydeadd** [hiddify/hiddify-app#864] [Ideas]
### easy way :
1. install default version from [https://github.com/hiddify/hiddify-next/releases/download/v1.1.1/Hiddify-MacOS.dmg](url)
2. download and unzip this file : [Hiddify Admin.zip](https://github.com/hiddify/hiddify-next/files/15255759/Hiddify.Admin.zip)
3. press command+space and type " script editor " and press enter .
4. locate the unzipped file and open it .
5. type your administrator username and password in " " .
6. press command+s and close script editor .
7. move this file to your application folder and right click and click open .
8. now this is the hiddify app with service mode on VPN(experimental)

> if you’re configs list are empty or you can’t add new configs from clipboard , close this app and first using terminal and type 
sudo /Applications/Hiddify.app/Contents/MacOS/Hiddify  , add your configs and close the app , then u can see your config list in Hiddify Admin .

### script editor code : 
`do shell script "sudo /Applications/Hiddify.app/Contents/MacOS/Hiddify" user name "username" password "password" with administrator privileges`

**Comment by ghost** [comment]
Also you can create Apple script with Automator app, then place it in Applications.

do shell script "/Applications/Hiddify.app/Contents/MacOS/Hiddify" with administrator privileges

**Comment by Act0r1** [comment]
Is this way actual ? Cause here no hiddify admin zip file

**Reply by elixirvitae** [reply]
Sure. I've just cried out all above, including Terminal command and it perfectly works (vpnd.io tariff)

**Comment by rekruizer** [comment]
Thank you for the application. The code inside app Hiddify Admin worked better for me `do shell script "sudo launchctl submit -l com.hiddify.app -- /Applications/Hiddify.app/Contents/MacOS/Hiddify" with administrator privileges`

Because it doesnt needs Hiddify admin to run at the same time with Hiddify. It just Runs Hiddify. 

Also, I prefer not to paste my password anywhere. So for me it's better to write my password every time I open hiddify admin

**Reply by Giant-Li** [reply]
Yes, your script needs password only at the first start. But now I can't close or quit the app, it always reopen automatically. And each time it reopens, it is a fresh start which means no configs in it.
Do you have this problem?

**Reply by rekruizer** [reply]
> Yes, your script needs password only at the first start. But now I can't close or quit the app, it always reopen automatically. And each time it reopens, it is a fresh start which means no configs in it.
> 
> Do you have this problem?

Sounds like a strange behavior for me. I don't have that issue.

Actually, tbh, I started using Happ app as my main VLESS vpn client. Sacrificing "auto" mode, that Hiddify has

**Reply by whoisfdv** [reply]
Thank you for the implementation.

I slightly modified the solution so that the client runs with administrator privileges while also executing within the user’s GUI session:

```
do shell script "sudo launchctl asuser $(stat -f %u /dev/console) /Applications/Hiddify.app/Contents/MacOS/Hiddify" user name "..." password "..." with administrator privileges
```
---
> Yes, your script needs password only at the first start. But now I can't close or quit the app, it always reopen automatically. And each time it reopens, it is a fresh start which means no configs in it.
>
> Do you have this problem?

Regarding the issue of the client restarting after being closed: `launchctl submit` creates a launchd daemon that incorrectly handles client termination and continuously respawns it as a new PID process.

The solution using `launchctl asuser` described above also resolves this issue.

Applicable to macOS 26.3.1.

**Comment by propulsivo** [comment]
Script works the best without username and pass in there. This is grand! Thank you!
